back-end/db/models/relationships.js

```diff
@@ -5,50 +5,27 @@ const Class = require("./classes");
 const ClassProgress = require("./classProgress");
 const AssignedBadge = require("./assignedBadges");
 
-Quest.belongsTo(User, {
-  foreignKey: "villager_id",
-  as: "villager",
-});
-
-Quest.belongsTo(User, {
-  foreignKey: "adventurer_id",
-  as: "adventurer",
-});
-
-Quest.belongsTo(Class, {
-  foreignKey: "class_id",
-});
-
-User.hasMany(Quest, {
-  foreignKey: "villager_id",
-  as: "villagerQuests",
-});
-
-User.hasMany(Quest, {
-  foreignKey: "adventurer_id",
-  as: "adventurerQuests",
-});
-
-ClassProgress.belongsTo(Class, {
-  foreignKey: "class_id",
-});
-
-ClassProgress.belongsTo(User, {
-  foreignKey: "adventurer_id",
-  as: "adventurer",
-});
+Quest.belongsTo(User, { as: "villager" });
+
+Quest.belongsTo(User, { as: "adventurer" });
+
+Quest.belongsTo(Class);
+
+User.hasMany(Quest, { as: "villagerQuests" });
```

Comment:
Here sequelize would be looking for a foreign key called `villagerQuestId` by default. We have no such thing, so we need to be explicit:

```js
User.hasMany(Quest, { foreignKey: "villagerId", as: "villagerQuests" });
```

Same story with "adventurerQuests".